torch/_dynamo/variables/builder.py

```diff
@@ -502,6 +495,12 @@ def index_source(key):
                 value,
                 guards=make_guards(GuardBuilder.FUNCTION_MATCH),
             )
+        elif issubclass(type(value), type):
+            # TODO(whc) the following seems preferable but breaks some tests, debug
+            # elif inspect.isclass(value):
+            return UserDefinedClassVariable(
+                value, guards=make_guards(GuardBuilder.FUNCTION_MATCH)
+            )
```

Comment:
The type of a class with metaclass is the metaclass rather than ```type```, but the type of metaclass is ```type```, so ```issubclass``` is a good match for this pattern. ```ABCMeta``` actually is one of the special case for metaclass, so don't need to special handling any more.
We need to move this down, because ```AutogradFunctionVariable``` uses metaclass to identify, so this pattern should be after that.